src/marqo/tensor_search/tensor_search.py

```diff
@@ -81,6 +81,47 @@
 logger = get_logger(__name__)
 
 
+def _get_dimension_from_model_properties(model_properties: dict):
+    """
+    Args:
+        model_properties: dict containing model properties
+    """
+    try:
+        return model_properties["dimensions"]
+    except KeyError:
+        raise errors.InvalidArgError(
```

Comment:
It's an `InvalidArgError` because this `KeyError` would occur when it's a custom model, meaning the user provided a `model_properties` dict, but it did not contain `dimensions`.